### ISSUE TITLE
Updating the HDF5 C library to supported HDF5 1.14.x series, JHDF5 compatibility

### ISSUE THREAD

**Issue by mkitti** [saalfeldlab/n5-hdf5#33]
Currently, n5-hdf5 is built on JHDF5 19.04.01. This contains [HDF5 1.10.5 binaries](https://sissource.ethz.ch/sispub/jhdf5/-/commit/d028d6121916273f0066673f7dfae7f919c3d60b) built in March 2019. One exception are the ARM macOS binaries from HDF5 1.10.9 built in [May 2022](https://sissource.ethz.ch/sispub/jhdf5/-/commit/ecc1cfb7a1f8dab428906c874e3e7b407b450f77) by @mzouink and I. JHDF5 has seen no updates since 2022.

According to The HDF Group release schedule HDF5 1.10.11 is the last release in the 1.10.x series. This makes it unlikely that JHDF5 will seen updates from the current maintainers. The current release is 1.14.3 with 1.14.4 due in a few months.

![image](https://github.com/saalfeldlab/n5-hdf5/assets/8062771/f719e81d-5f01-4bfe-8e6d-cc97970bb7f1)

I propose that we migrate to an actively maintained HDF5 dependency: [JavaCPP Presets for HDF5](https://mvnrepository.com/artifact/org.bytedeco/hdf5/1.14.3-1.5.10).

Via the JavaDocs for `org.bytedeco.hdf5` we can see that the package exposes both the `org.bytedeco.hdf5` automatically generated Java bindings for the C API and the `hdf5.hdflib` bindings from The HDF Group:
http://bytedeco.org/javacpp-presets/hdf5/apidocs/

To provide a bridge from existing usage of JHDF5 I rebased the JHDF5 API to depend on the APIs exposed by JavaCPP Presets for HDF5.
https://github.com/JaneliaSciComp/janelia-jhdf5

A current blocker is to resolve compilation issues on Windows:
https://github.com/bytedeco/javacpp-presets/issues/1405
Fixing the issue involves either convincing Visual Studio C++ to create a large DLL that exports all symbols or create multiple DLLs which depend on each other and which each expose parts of the API.

In summary, the above plan will
1) Upgrade the underlying HDF5 C library to version 1.14.3 or newer.
2) Expose three APIs for accessing the HDF5 library:
    i) `hdf.hdflib` API from The HDF Group
    ii) `org.bytedeco.hdf5` API from JavaCPP that is auto-generated from the C API
    iii) `ch/systemsx/cisd/hdf5` API from JHDF5, rebased on the above APIs.
